### PR TITLE
Fighter and UI Improvements

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
- Adds starmap button to fighter interface for FTL-capable fighters
- Fixes ammo display on fighter UI
- Adds armour readouts to the TAC console

## Why It's Good For The Game
QOL

## Changelog
:cl:
add: Added starmap button on pilot UI for sabres
add: Armour readouts on the tactical console, toggleable by clicking their names
fix: Fighter ammo display no longer stuck at zero
fix: Fighter target lock-on works
/:cl:

### CODE REVIEW

File: nsv13/code/game/general_quarters/dropship.dm [hunk 1]
```diff
@@ -329,14 +331,21 @@
...
+			OM.dradis?.ui_interact(usr)
 			return
 		if("toggle_ftl")
 			var/obj/item/fighter_component/ftl/ftl = OM.loadout.get_slot(HARDPOINT_SLOT_FTL)
 			if(!ftl)
 				return
 			ftl.active = !ftl.active
 			OM.relay('nsv13/sound/effects/fighters/switch.ogg')
+		if("show_starmap")
+			if(!OM?.starmap)
```
Comment (TheFakeElon) [inline nsv13/code/game/general_quarters/dropship.dm:343]: There's already a check for null `OM` on line 242 :)

File: nsv13/code/game/general_quarters/dropship_types.dm [hunk 2]
```diff
@@ -46,6 +46,20 @@ Credit to TGMC for the interior sprites for all these!
 						/obj/item/fighter_component/countermeasure_dispenser)
 	interior_mode = INTERIOR_DYNAMIC
 
+/obj/structure/overmap/fighter/dropship/Initialize(mapload, list/build_components)
+	. = ..()
```
Comment (TheFakeElon) [inline nsv13/code/game/general_quarters/dropship_types.dm:50]:
```suggestion
	return ..()
```

File: nsv13/code/game/general_quarters/dropship_types.dm [hunk 3]
```diff
@@ -46,6 +46,20 @@ Credit to TGMC for the interior sprites for all these!
...
+
+/obj/structure/overmap/fighter/dropship/post_load_interior()
+	var/obj/item/fighter_component/ftl/ftl = loadout.get_slot(HARDPOINT_SLOT_FTL)
+	if(ftl)
+		if(linked_areas?.len == 1)
```
Comment (TheFakeElon) [inline nsv13/code/game/general_quarters/dropship_types.dm:55]: Length can handle null arguments, faster too
```suggestion
		if(length(linked_areas) == 1)
```

**Review by Bokkiewokkie — DISMISSED** [review]

File: nsv13/code/modules/overmap/starmap.dm [hunk 4]
```diff
@@ -19,13 +19,32 @@
 	var/current_sector = 2
 	circuit = /obj/item/circuitboard/computer/ship/navigation
 
+/obj/machinery/computer/ship/navigation/can_interact(mob/user) //Override this code to allow people to use consoles when flying the ship.
+	if(locate(user) in linked?.operators)
```
Comment (TheFakeElon) [inline nsv13/code/modules/overmap/starmap.dm:23]: `locate()` with lists is usually used for locating types, if you're looking for a specific instance of a datum you can just use
```suggestion
	if(user in linked?.operators)
```

**Review by Karmic-Skink — APPROVED** [review]

**Comment by Kenionatus** [comment]
The following code in [starmap.dm](https://github.com/covertcorvid/NSV13/blob/24787c245988b1955d01363588e677b14a662318/nsv13/code/modules/overmap/starmap.dm) makes the navigation computer interactible in at any distance and in every state (e.g. in crit)
```
/obj/machinery/computer/ship/navigation/ui_state(mob/user)
	return GLOB.always_state
```
I suppose that was to allow interiorless ships to have an ftl drive, but we don't have those (any more) I think. I'm looking into fixing #2045. Do you think this can simple be set back to the default for tguis, @covertcorvid?

**Pull request merged**